plugins/UsersManager/Sql/UserTableFilter.php

```diff
@@ -107,11 +107,12 @@ private function getAccessSelectSqlCondition()
                 $sql = "u.superuser_access = 1";
                 break;
             default:
-                $sql = "a.access = ?";
+                $sql = "u.login IN (SELECT login from " . Common::prefixTable('access') . " WHERE access = ? AND (idsite IS NULL OR idsite = ?))";
                 $bind[] = $this->filterByRole;
+                $bind[] = $this->filterByRoleSite;
```

Comment:
this change is needed, as when filtering for a certain role/capability the result currently only contains the role/capability filtered for instead of having the `GROUP_CONCAT` of the main query containing all access levels for a user.